doc/devel/testing.rst

```diff
@@ -53,58 +56,72 @@ value.
 Running the tests
 -----------------
 
-Running the tests is simple. Make sure you have nose installed and run::
+Running the tests is simple. Make sure you have pytest installed and run::
+
+   py.test
+
+or::
 
    python tests.py
 
 in the root directory of the distribution. The script takes a set of
 commands, such as:
 
 ========================  ===========
-``--pep8``                pep8 checks
-``--no-pep8``             Do not perform pep8 checks
+``--pep8``                Perform pep8 checks (requires pytest-pep8_)
 ``--no-network``          Disable tests that require network access
 ========================  ===========
 
-Additional arguments are passed on to nosetests. See the nose
-documentation for supported arguments. Some of the more important ones are given
-here:
+Additional arguments are passed on to pytest. See the pytest documentation for
+supported arguments. Some of the more important ones are given here:
 
 =============================  ===========
 ``--verbose``                  Be more verbose
-``--processes=NUM``            Run tests in parallel over NUM processes
-``--process-timeout=SECONDS``  Set timeout for results from test runner process
-``--nocapture``                Do not capture stdout
+``--n NUM``                    Run tests in parallel over NUM
+                               processes (requires pytest-xdist_)
+``--timeout=SECONDS``          Set timeout for results from each test
+                               process (requires pytest-timeout_)
+``--capture=no`` or ``-s``     Do not capture stdout
 =============================  ===========
 
-To run a single test from the command line, you can provide a
-dot-separated path to the module followed by the function separated by
-a colon, e.g., (this is assuming the test is installed)::
+To run a single test from the command line, you can provide a dot-separated
+path to the module, optionally followed by the function separated by two
+colons, e.g., (this is assuming the test is installed)::
 
-  python tests.py matplotlib.tests.test_simplification:test_clipping
+  python tests.py matplotlib.tests.test_simplification::test_clipping
```

Comment:
I don't know; does anyone use this method?

On further though, using non-installed tests might be broken when trying `mpl_toolkits` tests because they import stuff from `matplotlib.tests`. I can move things around to get that working again.